### PR TITLE
 [DICT] Add Python support for clear 

### LINKED ISSUE #1880: [DICT] Add Python support for clear
### Describe the feature
Syntactic and semantic support should be added for `clear`. This should be tested by implementing Python printing

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
def foo():
    a = {1,1.0, 2:2.0}
    a.clear()
    return a
```

### PR DESCRIPTION
**Pull request by leonardopaloschi**
Files dict_methods.py, class_defs.py, test_epyccel_dicts.py and pycode.py were changed. Reference to issue #1880.
Closes #1880

### CODE REVIEW

File: tests/epyccel/test_epyccel_dicts.py [hunk 1]
```diff
@@ -181,3 +181,15 @@ def get_default_str_keys():
     python_result = get_default_str_keys()
     assert isinstance(python_result, type(pyccel_result))
     assert python_result == pyccel_result
+
+def test_dict_clear(language):
+    def dict_clear():
+        a = {1:1.0, 2:2.0}
+        a.clear()
+        return a
```
Comment (EmilyBourne) [inline tests/epyccel/test_epyccel_dicts.py:189]: You will need to use `len` to test in order to be able to test other languages (C/Fortran) as returning a dictionary is not yet supported for these languages
```suggestion
        a.clear()
        return len(a)
```

File: tests/epyccel/test_epyccel_dicts.py [hunk 2]
```diff
@@ -181,3 +181,15 @@ def get_default_str_keys():
...
+    epyc_dict_clear = epyccel(dict_clear, language = language)
+    pyccel_result = epyc_dict_clear()
+    python_result = dict_clear()
+    assert len(python_result) == len(pyccel_result)
+    assert len(python_result) == 0
+    assert len(pyccel_result) == 0
```
Comment (EmilyBourne) [inline tests/epyccel/test_epyccel_dicts.py:195]: You don't need to use `len` outside the function that is translated. You are already returning the dictionary (this is ok as you are only translating to Python) so you can simply do:
```suggestion
    assert python_result == pyccel_result
```

File: pyccel/codegen/printing/pycode.py [hunk 3]
```diff
@@ -854,6 +854,10 @@ def _print_DictGet(self, expr):
             return f"{dict_obj}.get({key}, {val})\n"
         else:
             return f"{dict_obj}.get({key})\n"
+        
+    def _print_DictClear(self, expr):
+        dict_obj = self._print(expr.dict_obj)
+        return f"{dict_obj}.clear()\n"
```
Comment (EmilyBourne) [inline pyccel/codegen/printing/pycode.py:860]: Are you sure that this is needed? I would expect `_print_DictMethod` to be sufficient

**Comment by EmilyBourne** [comment]
Please make sure to [link your pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue) so it will automatically be closed when the PR is merged.

**Comment by leonardopaloschi** [comment]
Hi @EmilyBourne,

I’ve made all the changes you requested. I deleted the print method, updated the assert, and changed the _class_type. The only thing I’m unsure about is the type; I committed it with InhomogeneousTupleType(), but I’m not sure if it should be VoidType() in that context.

Let me know what you think! If everything looks good, please let me know what else I need to do.

Thanks!

**Comment by EmilyBourne** [comment]
I think `VoidType` is probably right. The type describes the type of the result of the function. So the type of `a` in:
```python
def f(b : dict[int, float]):
    a = b.clear()
```
`VoidType()` is used if the function doesn't return anything. `InhomogeneousTupleType` would be the type of `()`.

Please check with the bot for the next steps. I will leave a complete review when the bot requests it

**Comment by leonardopaloschi** [comment]
Hi, Emily.
It seems that all the tests passed except for the documentation one. However, the error reported is related to the docstring of a part of the code that apparently has no connection to what I did, so I would like to know what I can do to have the PR reviewed. Thank you in advance.

**Comment by EmilyBourne** [comment]
@leonardopaloschi the documentation test compares with the devel branch. As your branch is not up to date with devel it sees docs that have been added on devel as things that have been removed on your branch. Please merge devel into your branch and fix the conflicts. This should fix the problem

File: pyccel/ast/builtin_methods/dict_methods.py [hunk 4]
```diff
@@ -175,5 +177,22 @@ def default_value(self):
         The value that should be returned if the key is not present in the dictionary.
         """
         return self._args[1]
+#==============================================================================
+class DictClear(DictMethod) :
```
Comment (EmilyBourne) [inline pyccel/ast/builtin_methods/dict_methods.py:181]: As per PEP: https://peps.python.org/pep-0008/#blank-lines
> Surround top-level function and class definitions with two blank lines.

```suggestion
        return self._args[1]

#==============================================================================
class DictClear(DictMethod) :
```

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
This looks good. I just have a couple of minor comments.
I see that you have ticked the items corresponding to the AUTHORS and the CHANGELOG in your checklist but I don't see these changes. Please can you fix this

**Review by EmilyBourne — APPROVED** [review]
This looks good. Thank you for these changes

File: tests/epyccel/test_epyccel_dicts.py [hunk 5]
```diff
@@ -204,3 +204,13 @@ def dict_ptr():
...
+
+def test_dict_clear(python_only_language):
+    def dict_clear():
+        a = {1:1.0, 2:2.0}
+        a.clear()
+        return a
+    epyc_dict_clear = epyccel(dict_clear, language = python_only_language)
+    pyccel_result = epyc_dict_clear()
+    python_result = dict_clear()
+    assert python_result == pyccel_result
```
Comment (EmilyBourne) [inline tests/epyccel/test_epyccel_dicts.py:216]:
```suggestion
    assert python_result == pyccel_result

```

File: pyccel/ast/class_defs.py [hunk 6]
```diff
@@ -11,7 +11,7 @@
 from pyccel.ast.builtin_methods.list_methods import (ListAppend, ListInsert, ListPop,
                                                      ListClear, ListExtend, ListRemove,
                                                      ListCopy, ListSort)
-from pyccel.ast.builtin_methods.dict_methods  import( DictPop, DictPopitem, DictGet,
+from pyccel.ast.builtin_methods.dict_methods  import DictPop, DictPopitem, DictGet, DictClear,
```
Comment (EmilyBourne) [inline pyccel/ast/class_defs.py:14]:
```suggestion
from pyccel.ast.builtin_methods.dict_methods  import (DictPop, DictPopitem, DictGet, DictClear,
```

File: pyccel/ast/class_defs.py [hunk 7]
```diff
@@ -11,7 +11,7 @@
 from pyccel.ast.builtin_methods.list_methods import (ListAppend, ListInsert, ListPop,
                                                      ListClear, ListExtend, ListRemove,
                                                      ListCopy, ListSort)
-from pyccel.ast.builtin_methods.dict_methods  import( DictPop, DictPopitem, DictGet,
+from pyccel.ast.builtin_methods.dict_methods  import (DictPop, DictPopitem, DictGet, DictClear,
```
Comment (EmilyBourne) [inline pyccel/ast/class_defs.py:14]:
```suggestion
from pyccel.ast.builtin_methods.dict_methods import (DictPop, DictPopitem, DictGet, DictClear,
```

**Pull request merged**